flax/linen/linear.py

```diff
@@ -45,6 +46,10 @@
   LaxPadding,
 )
 
+class PromoteDtypeFn(Protocol):
```

Comment:
why are there two copies of the protocol?  I see another in typing.py

Reply:
Yeah they are slightly different because the version that Linen users is a bit less `typing` friendly.